features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/utils/PathUtils.kt

```diff
@@ -0,0 +1,186 @@
+            logger.log(
+                target = InternalLogger.Target.MAINTAINER,
+                level = InternalLogger.Level.WARN,
+                messageBuilder = { PATH_CONVERSION_ERROR },
+                throwable = e
+            )
+            null
+        }
+    }
+
+    internal fun convertPathToBitmap(
+        checkPath: Path,
+        fillColor: Int,
+        checkmarkColor: Int
+    ): Bitmap? {
+        val scaledPath = scalePathToBitmapSize(checkPath)
+        val mutableBitmap =
+            bitmapWrapper.createBitmap(CHECKBOX_SIZE_DP, CHECKBOX_SIZE_DP, Bitmap.Config.ARGB_8888)
+                ?: return null
+
+        return drawPathOntoBitmap(
+            mutableBitmap,
+            scaledPath,
+            fillColor,
+            checkmarkColor
+        )
+    }
+
+    private fun drawPathToBitmap(checkmarkColor: Int, path: Path, canvas: Canvas?) {
+        val paint = Paint().apply {
+            color = checkmarkColor
+            style = Paint.Style.STROKE
+            strokeWidth = STROKE_WIDTH_DP
+            isAntiAlias = true
+        }
+
+        // Draw the Path onto the Canvas
+        asAndroidPathSafe(path)?.let {
+            drawPathSafe(canvas, it, paint)
+        }
+    }
+
+    private fun scalePathToBitmapSize(path: Path): Path {
+        // path initial bounds
+        val originalBounds = path.getBounds()
+
+        // calculate the scale factor
+        val scaleX = CHECKBOX_SIZE_DP / originalBounds.width
+        val scaleY = CHECKBOX_SIZE_DP / originalBounds.height
+        val scaleFactor = minOf(scaleX, scaleY)
+
+        // current center
+        val currentCenterX = (originalBounds.left + originalBounds.right) / 2
+        val currentCenterY = (originalBounds.top + originalBounds.bottom) / 2
+
+        // new center
+        val newCenterX = CHECKBOX_SIZE_DP / 2
+        val newCenterY = CHECKBOX_SIZE_DP / 2
+
+        // center changes after scaling
+        val scaledCenterX = currentCenterX * scaleFactor
+        val scaledCenterY = currentCenterY * scaleFactor
+
+        // translation needed to recenter
+        val translateX = newCenterX - scaledCenterX
+        val translateY = newCenterY - scaledCenterY
+
+        // the order of operations is important
+        val matrix = Matrix()
+        matrix.translate(translateX, translateY)
+        matrix.scale(scaleFactor, scaleFactor)
+        path.transform(matrix)
+
+        return path
+    }
+
+    private fun drawPathOntoBitmap(
+        bitmap: Bitmap,
+        scaledPath: Path,
+        fillColor: Int,
+        checkmarkColor: Int
+    ): Bitmap? {
+        val canvas = canvasWrapper.createCanvas(bitmap) ?: return null
+        drawCanvasBackground(canvas, fillColor)
+        drawCanvasForeground(canvas, scaledPath, checkmarkColor)
+        return bitmap
+    }
+
+    private fun drawCanvasBackground(
+        canvas: Canvas,
+        fillColor: Int
+    ) {
+        canvas.drawColor(fillColor)
+    }
```

Comment:
seems like no need for the dedicated method? same for `drawCanvasForeground`. You can rather add a text comment if clarity is needed.